apps/api/views.py

```diff
@@ -548,6 +548,10 @@ def trigger_bot_message(request):
 
     if platform == ChannelPlatform.COMMCARE_CONNECT and not participant_data.has_consented():
         return JsonResponse({"detail": "User has not given consent"}, status=status.HTTP_400_BAD_REQUEST)
+    if participant_data and data["participant_data"]:
```

Comment:
I know the serializer has `default=dict` but I would still use `data.get("participant_data")` here for safety.

Also, if the request includes participant data but the DB object doesn't exist, we should create one.